src/java/org/apache/cassandra/tools/nodetool/GuardrailsConfigCommand.java

```diff
@@ -112,89 +175,96 @@ public static class SetGuardrailsConfig extends GuardrailsConfigCommand
                    .sorted(comparing(Method::getName))
-                   .collect(toList());
+                   .collect(Collectors.groupingBy(method -> toSnakeCase(method.getName().substring(3))))
+                   .entrySet()
+                   .stream()
+                   .sorted(Map.Entry.comparingByKey())
+                   .collect(Collectors.toMap(Map.Entry::getKey,
+                                             Map.Entry::getValue,
+                                             (e1, e2) -> e1,
+                                             LinkedHashMap::new));
         }
 
         private String sanitizeSetterName(Method setter)
         {
             return toSnakeCase(SETTER_PATTERN.matcher(setter.getName()).replaceAll(""));
         }
 
-        private void executeSetter(NodeProbe nodeProbe)
+        private void sanitizeArguments(Method setter, List<String> args)
         {
-            String snakeCaseName = args.get(0);
-            String setterName = toCamelCase(args.get(0).startsWith("set_") ? args.get(0) : "set_" + args.get(0));
-
-            Method setter = getAllSetters(nodeProbe).stream()
-                                                    .findFirst()
-                                                    .orElseThrow(() -> new IllegalStateException(format("Setter method %s not found. " +
-                                                                                                        "Run nodetool setguardrailsconfig --list " +
-                                                                                                        "to see available setters", setterName)));
-
-            validateArguments(setter, snakeCaseName, args);
-
-            List<String> methodArgs = args.subList(1, args.size());
-            try
+            Class<?>[] parameterTypes = setter.getParameterTypes();
+            if (parameterTypes.length == 1 && parameterTypes[0] == Set.class)
             {
-                setter.invoke(nodeProbe.getGuardrailsMBean(), prepareArguments(methodArgs, setter));
-            }
-            catch (Exception ex)
-            {
-                String reason;
-                if (ex.getCause() != null && ex.getCause().getMessage() != null)
-                    reason = ex.getCause().getMessage();
-                else
-                    reason = ex.getMessage();
-
-                throw new IllegalStateException(format("Error occured when setting the config for setter %s with arguments %s: %s",
-                                                       snakeCaseName, methodArgs, reason));
+                if (args.size() > 2)
+                {
+                    String guardrail = args.get(0);
+                    // replace multiple arguments with one which is separated by a single comma
+                    String collectedArguments = String.join(",", args.subList(1, args.size()));
+                    args.clear();
```

Comment:
`args.clear();` - this could cause problems for someone in the future who wants to debug the command because the input has magically changed inside one of the methods, making it difficult to find (and read). I would suggest using another array, leaving the input array as is or assing `guardrailName` and/or `guardrailValue` to a different variables.

Anyway this is a nit.

Reply:
Yes I am aware of args being changed and I actually wanted it like that. The method is actually called like that ... "sanitize arguments" so it is expected the args will be changed after that method is done.